polygeist/lib/polygeist/Passes/ConvertPolygeistToLLVM.cpp

```diff
@@ -45,16 +49,513 @@
 using namespace mlir;
 using namespace polygeist;
 
-mlir::LLVM::LLVMFuncOp GetOrCreateMallocFunction(ModuleOp module);
-mlir::LLVM::LLVMFuncOp GetOrCreateFreeFunction(ModuleOp module);
+/// Conversion similar to the canonical one, but not inserting the obtained
+/// pointer in a struct.
+struct GetGlobalMemrefOpLowering
+    : public ConvertOpToLLVMPattern<memref::GetGlobalOp> {
+  using ConvertOpToLLVMPattern<memref::GetGlobalOp>::ConvertOpToLLVMPattern;
+
+  LogicalResult
+  matchAndRewrite(memref::GetGlobalOp getGlobalOp, OpAdaptor adaptor,
+                  ConversionPatternRewriter &rewriter) const override {
+    const auto memrefTy = getGlobalOp.getType();
+    if (!canBeLoweredToBarePtr(memrefTy))
+      return failure();
+
+    const auto loc = getGlobalOp.getLoc();
+    const auto arrayTy =
+        convertGlobalMemrefTypeToLLVM(memrefTy, *typeConverter);
+    if (!arrayTy)
+      return failure();
+    const auto memSpace = memrefTy.getMemorySpaceAsInt();
```

Comment:
[nit]: similar comment as above, no need to create memSpace variable given that there is a single use.